nncf/quantization/helpers.py

```diff
@@ -0,0 +1,60 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Optional
+
+from nncf.api.compression import ModelType
+from nncf.common.utils.backend import get_backend
+from nncf.common.utils.backend import BackendType
+from nncf.common.quantization.structs import QuantizationPreset
+from nncf.data import Dataset
+from nncf.quantization.params import TargetDevice
+
+
+def quantize(model: ModelType,
+             calibration_dataset: Dataset,
+             preset: QuantizationPreset = QuantizationPreset.PERFORMANCE,
+             target_device: TargetDevice = TargetDevice.ANY,
+             subset_size: int = 300,
+             fast_error_correction: bool = True,
+             model_type: Optional[str] = None) -> ModelType:
```

Comment:
Need to think really hard on how we will expose this function and with which signature.

Reply:
Thanks for the comment! It was discussed with @AlexKoff88. The proposal was implemented here. We would like to use this function as `nncf.quantize()`  with this list of parameters.

Reply:
We will extend the list of parameters in the future. Probably, we should add the `advanced_parameters: AdvancedQuantizationParameters` optional argument to support features as ignored scope, some inference parameters and etc. We will discuss the structure of the `AdvancedQuantizationParameters` and introduce this in future PRs.